lib/google/ads/google_ads/logging_interceptor.rb

```diff
@@ -41,8 +41,21 @@ def request_response(request:, call:, method:, metadata: {})
                 call.instance_variable_get('@wrapped').peer,
                 method
               )
-          request_message = sprintf("Outgoing request: Headers: %s Payload: %s",
-              metadata.to_json, request.to_json)
+
+          # calling #to_json on some protos (specifically those with non-UTF8
+          # encodable byte values) causes a segfault, however #inspect works
+          # so we check if the proto contains a bytevalue, and if it does
+          # we #inspect instead of #to_json
+          request_inspect = if /Google::Protobuf::BytesValue/ === request.inspect
+            request.inspect
```

Comment:
Can we avoid doing "request.inspect" twice in a row if it does wind up including that regex? It's probably not that expensive but this happens every API call.

Reply:
You're still actually calling .inspect twice. You should re-use ri inside the if statement.

I wonder if there's a faster way to do this check than a regex also?